blocks/header/header.js

```diff
@@ -88,8 +140,65 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   } else {
     window.removeEventListener('keydown', closeOnEscape);
   }
+
+  const navMenuOverlay = navSections.querySelector(
+    Constants.NAV_MENU_OVERLAY_WITH_SELECTOR,
+  );
+  if (!expanded) {
+    navMenuOverlay.classList.add(Constants.OPEN);
+  } else {
+    navMenuOverlay.classList.remove(Constants.OPEN);
+  }
+
+  // enable menu collapse on escape keypress
+  if (!expanded || isDesktop.matches) {
+    // collapse menu on escape press
+    window.addEventListener(Constants.KEY_DOWN, closeOnEscape);
+  } else {
+    window.removeEventListener(Constants.KEY_DOWN, closeOnEscape);
+  }
 }
+function formatNavigationJsonData(navJson) {
+  const structuredData = [];
+  let currentLevel0 = null;
+  let currentCategory = null;
+  let currentLevel1 = null;
```

Comment:
I think you can do 
```js
let currentLevel0, currentCategory, currentLevel1;
```
That will initialize the variable `undefined` until you assign it a value.